test_test.go

```diff
@@ -20,7 +20,7 @@ type SampleObject struct {
 func init() {
 	api = httptest.NewServer(
 		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			if r.Method == "POST" {
+			if r.Method == "POST" || r.Method == "PUT" {
```

Comment:
no need to change this, but I believe in Go 1.7 there were constants added for the HTTP methods, i.e. `http.MethodPut` .